src/components/dashboard/components/SearchBar.jsx

```diff
@@ -3,115 +3,92 @@ import React from 'react';
-            <i className="fas fa-search fa-fw mr-1 col-1" />
+    <div className="row justify-content-center search-container">
+      <div className="concept-search-wrapper col-12 col-md-5 col-sm-8">
+        <i className="fas fa-search fa-fw mr-1" />
+        <form id="submit-search-form" onSubmit={onSubmit}>
+          <input
+            type="search"
+            name="searchInput"
+            id="search"
+            value={searchValue}
+            onChange={onSearch}
+            className="concept-search"
+            placeholder="Search for a source"
+          />
+        </form>
+      </div>
+      <div className="col-0" />
+      <div className="concept-search-wrapper dropdown d-none d-md-inline d-lg-inline d-xl-inline mini-sort-btn col-md-2">
+        <a
+          className="dropdown-toggle"
+          href="#"
+          role="button"
+          id="sourceType"
+          data-toggle="dropdown"
+          aria-haspopup="true"
+          aria-expanded="false"
+        >
+          Filter by source type
+          </a>
+        <div className="dropdown-menu source-menu" aria-labelledby="sourceType">
+          <div className="form-check ml-3">
             <input
-              type="search"
-              name="searchInput"
-              id="search"
-              value={searchValue}
+              className="form-check-input"
+              type="checkbox"
+              name="dictionary"
+              id="dictionary"
               onChange={onSearch}
-              className="search-bar col-8 col-sm-8 col-md-7 col-lg-8"
-              placeholder="Search for a source"
             />
-            <div className="dropdown d-inline d-md-none col-1 mini-source-btn">
-              <a
-                className="dropdown-toggle"
-                href="#"
-                role="button"
-                id="sources"
-                data-toggle="dropdown"
-                aria-haspopup="true"
-                aria-expanded="false"
-                data-tip="sources"
-              >
-                <i className="fas fa-list fa-fw" />
-              </a>
-
-              <div className="dropdown-menu" aria-labelledby="sources">
-                <a className="dropdown-item" href="#">
-                  Sources
-                </a>
-              </div>
-            </div>
-            <div className="dropdown d-none d-md-inline d-lg-inline d-xl-inline mini-sort-btn col-md-2">
-              <a
-                className="dropdown-toggle"
-                href="#"
-                role="button"
-                id="sourceType"
-                data-toggle="dropdown"
-                aria-haspopup="true"
-                aria-expanded="false"
-              >
-                Filter by source type
+            <label className="form-check-label" htmlFor="dictionary">
+              Dictionary
+              </label>
+          </div>
+          <div className="form-check ml-3">
```

Comment:
You have repeated the same div element `<div className="form-check ml-3">` about 4 times while only changing the attribute values. Would creating a single functional component that returns the same elements and takes in the attributes that vary as props be a cleaner method or does it also result in the same lines of code? Otherwise, this looks good to me.

Reply:
Using a functional component would have been a great idea, but I decided to re-use the div element in order to avoid breaking the app since this was just an enhancement to how the search bar UI looked before.